endpointmanager/pkg/fetcher/fhirlist.go

```diff
@@ -0,0 +1,67 @@
+package fetcher
+
+import (
+	log "github.com/sirupsen/logrus"
+)
+
+// FHIRList implements the Endpoints interface for an endpoint list in FHIR
+type FHIRList struct{}
+
+// GetEndpoints takes the list of endpoints in FHIR format and formats it into a ListOfEndpoints
+// Assumed Structure:
+/**
+{ ... entry: [ {
+		fullUrl: URI for resource
+		resource: {
+			...
+			name: <name of the endpoint>
+			managingOrganiation: { display: <text for resource>, reference: <organization name> },
+			address: <FHIR url>,
+		}
+	  }, ...
+] }
+*/
+func (fl FHIRList) GetEndpoints(fhirList []map[string]interface{}, listURL string) ListOfEndpoints {
+	var finalList ListOfEndpoints
+	var innerList []EndpointEntry
+
+	for entry := range fhirList {
+		fhirEntry := EndpointEntry{}
+		if listURL != "" {
+			fhirEntry.ListSource = listURL
+		} else {
+			fhirEntry.ListSource = "FHIR"
+		}
+
+		resource, ok := fhirList[entry]["resource"].(map[string]interface{})
+		if ok {
+			// Save both name & managing organization in the array since both could be used
+			// for storing the organization name
+			managingOrg, orgOk := resource["managingOrganization"].(map[string]interface{})
+			if orgOk {
+				orgName, orgOk := managingOrg["display"].(string)
+				if orgOk {
+					fhirEntry.OrganizationNames = append(fhirEntry.OrganizationNames, orgName)
+				}
+				alternateName, orgOk := managingOrg["reference"].(string)
+				if orgOk {
+					fhirEntry.OrganizationNames = append(fhirEntry.OrganizationNames, alternateName)
+				}
+			}
+			nameEndpt, nameOk := resource["name"].(string)
+			if nameOk {
+				fhirEntry.OrganizationNames = append(fhirEntry.OrganizationNames, nameEndpt)
+			}
+			uri, uriOk := resource["address"].(string)
+			if uriOk {
+				fhirEntry.FHIRPatientFacingURI = uri
+			}
+			innerList = append(innerList, fhirEntry)
```

Comment:
Just realizing this -- I don't think we should proceed to add entries to the "innerList" unless they've been successfully populated. For example we don't want to end up having an entry that doesn't have a URL

Reply:
that's fair! what would you consider "successfully populated" besides having the URL? does it also need to have the name field and/or the managingOrganization field? or do you think it just needs to have a URL at a minimum?